packages/core/src/wrapper/DeviceData.ts

```diff
@@ -0,0 +1,21 @@
+import { isCordovaApp, isReactNative } from './PlatformHelpers';
+
+// [Investigation] Detecting platforms on runtime.
+
+// Welcome to TypeScript hacks
+declare var device: any;
+// Welcome to TypeScript hacks
+declare var Platform: any;
+
+/**
+ * Get device platform
+ */
+export function getPlatform(): string {
+    if (isCordovaApp) {
+      // https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-device/
+      return device.platform;
+    } else if (isReactNative) {
+      return Platform.OS;
+    }
+    return 'Unknown';
+}
```

Comment:
-1 on this approach. It's not really reusable, and hard to extend. 

A better approach I think we can use is we should take advantage of the interfaces feature provided by TS, and define a common interface to provide the information that we need, and then provide different implementation backed by Cordova or ReactiveNative.

In this case, we can have:

```
interface Device {
  string getPlatform();
}

class CordovaDevice {
  string getPlatform() {
    return device.platform
  }
}

class ReactNativeDevice {
  string getPlatform() {
    return Platform.OS;
  }
}
```

Then during the init time, the core module will only need to detect the platform once, and then instantiate the right implementation automatically.

Reply:
Great! I think the same. I have done something like this alternative aproach: https://github.com/aerogear/aerogear-js-sdk/pull/16

What I think now is that we need something in the middle - we do not need separate module as in PR above, but we need to have wrappers and clasess as you have suggested.